### PR TITLE
Fixes RODEX retrieve all button

### LINKED ISSUE #5061: RoDEX 'Retriev all' function not working propertly
* **rAthena Hash**: https://github.com/rathena/rathena/commit/8efd61f9b27dd5163d8f71f55f2025470f1254b3
* **Client Date**: 20200401
* **Server Mode**: RE
* **Description of Issue**: It is not possible to retrieve all items present in the rodex using the 'retrieve all' button.
  * How to Reproduce: Accumulate various rewards on the rodex, click on the 'retrieve all' button, you will not receive any items.

Now, relog in the game, open the rodex and press 'retrieve all' again, you will crash the game.
But when you login again the items will be in your inventory.

'Delete all' button its working!

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: Fixes #5061

* **Server Mode**: Renewal

* **Description of Pull Request**: 
  * Resolves the client crashing when attempting to retrieve all mail that contain an item.
Thanks to @admkakaroto!

### CODE REVIEW

**Comment by 24msz6eo** [comment]
After testing, it will still crash. Client 20200401

**Comment by 24msz6eo** [comment]
After testing, you have to log in to the game again to receive attachments

**Comment by Balferian** [comment]
Client crashed when take items from **one** mail.
image

**Comment by 24msz6eo** [comment]
Following this pr modification, there is still a crash problem

File: src/char/int_mail.cpp [hunk 1]
```diff
@@ -361,9 +361,6 @@ void mapif_Mail_getattach(int fd, uint32 char_id, int mail_id, int type)
 	if( msg.dest_id != char_id )
 		return;
 
-	if( msg.status != MAIL_READ )
```
Comment (Lemongrass3110) [inline src/char/int_mail.cpp:364]: This should be PACKETVER dependent!
Although I do not know when this new button was added. :-(
But before that button a player would not have been able to get attachments from a mail without opening it, that's basically why that check existed.
Another option would be offering a config to let admins choose if they allow it or not?

File: src/map/clif.cpp [hunk 2]
```diff
@@ -15347,12 +15347,15 @@ void clif_mail_getattachment(struct map_session_data* sd, struct mail_message *m
 			return;
 	}
 
-	WFIFOHEAD(fd, 12);
-	WFIFOW(fd, 0) = type == MAIL_ATT_ZENY ? 0x9f2 : 0x9f4;
-	WFIFOQ(fd, 2) = msg->id;
-	WFIFOB(fd, 10) = msg->type;
-	WFIFOB(fd, 11) = result;
-	WFIFOSET(fd, 12);
+	PACKET_ZC_ACK_ITEM_FROM_MAIL *p = { 0 };
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:15350]: As discussed please change this from pointer to a real struct and use clif_send.
This also does not contain any logic change, but is only an update to new structure.

**Comment by Balferian** [comment]
@Lemongrass3110 This buttons added in ~2019-05-20.
https://ro.gnjoy.com/news/notice/View.asp?BBSMode=10001&seq=7117&curpage=1

> - 로덱스 시스템에 일부 기능이 추가됩니다.
> 　→ 모두받기, 선택받기, 모두삭제, 선택삭제

Some features will be added to your RODEX system.
Receive all, Receive selection, Delete all, Delete selected

**Comment by 24msz6eo** [comment]
Thanks to aleos89 contribution, the crash problem is still not solved

**Comment by aleos89** [comment]
The crash is with single check outs, I'm aware. But I haven't found a solution and it's possible it's related to the client date. I haven't had time to look deeper.

**Comment by attackjom** [comment]
Test in KRO when press '  Recieve all ' .  All item will got without 'Read Email' icon.

My fixed:   make clif_parse_mail_getattach  untouch the itemarray part in clif_read_email  by use some flag to not read them.

**Comment by 24msz6eo** [comment]
All checkpoint buttons lack negative weight judgment. Can this be solved?

**Pull request merged**